docs/tutorials/building_with_kaniko.md

```diff
@@ -0,0 +1,127 @@
+
+# Building with Kaniko
+
+Before starting, make sure you have a running:
+
+- Tekton deployment
+- Build deployment
+
+See the [`Try It!`](/README.md#Try_It!) section for more information.
+
+## Getting Started
+
+### Registry Authentication
+
+For this tutorial, we will require to create a `tutorial-secret` Kubernetes secret to access a [DockerHub](https://hub.docker.com/) registry, as follows:
+
+```sh
+$ REGISTRY_SERVER=https://index.docker.io/v1/ REGISTRY_USER=<your_registry_user> REGISTRY_PASSWORD=<your_registry_password>
+$ kubectl create secret docker-registry tutorial-secret --docker-server=$REGISTRY_SERVER --docker-username=$REGISTRY_USER --docker-password=$REGISTRY_PASSWORD  --docker-email=me@here.com
+```
+
+_Note_: For more information about authentication, please refer to the related [docs](/docs/development/authentication.md).
+
+## Create the strategy
+
+Ensure all strategies are in place, see the [`Try It!`](/README.md#Try_It!) section for more information.
+
+```sh
+$ kubectl get cbs
+NAME              AGE
+buildah           2m
+buildpacks-v3     2m
+kaniko            2m
+ko                2m
+source-to-image   2m
+```
+
+_Note_: For more information about strategies, please refer to the related [docs](/docs/buildstrategies.md).
+
+## Creating a Build
+
+For the Build definition, we will require the following:
+
+- A github repository containing a go [application](https://github.com/shipwright-io/sample-go/tree/main/docker-build) that requires a `Dockerfile`.
+- The `tutorial-secret` we just created.
+- The `kaniko` ClusterBuildStrategy.
+
+Let´s apply our Build and wait for it to be ready:
+
+```bash
+$ export REGISTRY_ORG=<your_registry_org>
+$ cat <<EOF | kubectl apply -f -
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: go-tutorial
+spec:
+  source:
+    url: https://github.com/shipwright-io/sample-go
+    contextDir: docker-build
+  strategy:
+    name: kaniko
+    kind: ClusterBuildStrategy
+  output:
+    image: docker.io/${REGISTRY_ORG}/go-tutorial:latest
+    credentials:
+      name: tutorial-secret
+EOF
+```
+
+Verify that the `go-tutorial` Build was created successfully:
+
+```sh
+kubectl get build
+NAME          REGISTERED   REASON      BUILDSTRATEGYKIND      BUILDSTRATEGYNAME   CREATIONTIME
+go-tutorial   True         Succeeded   ClusterBuildStrategy   kaniko              13s
+```
+
+_Note_: For more information about Build resources, please refer to the related [docs](/docs/build.md).
+
+## Creating a BuildRun
+
+Second, we will create a `BuildRun` resource that references our previous `go-tutorial` Build:
+
+```sh
+$ cat <<EOF | kubectl create -f -
+apiVersion: shipwright.io/v1alpha1
+kind: BuildRun
+metadata:
+  name: go-tutorial-buildrun
```

Comment:
```suggestion
  generateName: go-tutorial-buildrun-
```

Reply:
This was on purpose, otherwise it complicates the usage of `kubectl get buildrun go-tutorial-buildrun -o json | jq '.status.conditions[]'` command.